ApiDoctor.Console/Program.cs

```diff
@@ -3039,32 +3039,41 @@ private static CodeBlockAnnotation ExtractCodeBlockAnnotationForPermissionsTable
             return null;
         }
 
-        private static string ConvertToThreeColumnPermissionsTable(IEnumerable<string> tableRows)
+        private static string ConvertToThreeColumnPermissionsTable(IEnumerable<string> tableRows, string fileName)
         {
             var tableString = new StringBuilder("|Permission type|Least privileged permissions|Higher privileged permissions|");
             tableString.Append("\r\n|:---|:---|:---|");
             foreach (string row in tableRows)
             {
-                string[] cells = Regex.Split(row.Trim(), @"\s*\|\s*").Where(static x => !string.IsNullOrWhiteSpace(x)).ToArray();
-                
-                // We already have the 3 column permissions table, abort
-                if (cells.Length == 3)
+                try
                 {
-                    return string.Join("\r\n", tableRows);
-                }
+                    string[] cells = Regex.Split(row.Trim(), @"\s*\|\s*").Where(static x => !string.IsNullOrWhiteSpace(x)).ToArray();
```

Comment:
If we can, we should move the regex to static varaible. 
https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-regex#static-regular-expressions